functions/go/set-annotation/Dockerfile

```diff
@@ -1,8 +1,15 @@
 FROM golang:1.15-alpine3.12
 ENV CGO_ENABLED=0
 WORKDIR /go/src/
+
+COPY go.mod go.sum ./
```

Comment:
I don't think this can avoid duplicated downloading because each function is in a different container.

Reply:
To be clear, this is optimized for the following workflow for a single container:
- go build
- change go source code
- go build again

Before:
Because we use `COPY . .`, if we update anything (e.g. help text in main.go), the cache for this layer will not be reused. It will download all go modules again.

After:
`go mod download` is in a earlier layer and cache can be reused if we don't change `go.mod` and `go.sum`.

It will be 10+s faster on subsequent builds.